spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/LocalAppDeployer.java

```diff
@@ -159,10 +159,6 @@ public AppStatus status(String id) {
 			}
 		}
 		AppStatus status = builder.build();
-		// Make sure to explicitly undeploy any exited instances that have status undeployed.
-		if (DeploymentState.undeployed.equals(status.getState())) {
```

Comment:
I think this specific code was the result of handling `exit == 0` case which set the deployment status to `undeployed`.